src/Functions/UTCTimestampTransform.cpp

```diff
@@ -0,0 +1,169 @@
+
+        size_t getNumberOfArguments() const override { return 2; }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.size() != 2)
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {}'s arguments number must be 2.", name);
+            WhichDataType which_type_first(arguments[0]);
+            WhichDataType which_type_second(arguments[1]);
+            if (!which_type_first.isDateTime() && !which_type_first.isDateTime64())
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument type must be datetime.", name);
+            if (dynamic_cast<const TimezoneMixin *>(arguments[0].get())->hasExplicitTimeZone())
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument should not have explicit time zone.", name);
+            if (!which_type_second.isString())
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 2nd argument type must be string.", name);
+            DataTypePtr date_time_type;
+            if (which_type_first.isDateTime())
+                date_time_type = std::make_shared<DataTypeDateTime>();
+            else
+            {
+                const DataTypeDateTime64 * date_time_64 = static_cast<const DataTypeDateTime64 *>(arguments[0].get());
+                date_time_type = std::make_shared<DataTypeDateTime64>(date_time_64->getScale());
+            }
+            return date_time_type;
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t) const override
+        {
+            if (arguments.size() != 2)
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {}'s arguments number must be 2.", name);
+            ColumnWithTypeAndName arg1 = arguments[0];
+            ColumnWithTypeAndName arg2 = arguments[1];
+            const auto *  time_zone_col = checkAndGetColumn<ColumnString>(arg2.column.get());
+            if (!time_zone_col)
+            {
+                const auto * time_zone_const_col = checkAndGetColumn<ColumnConst>(arg2.column.get());
+                if (!time_zone_const_col)
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 2nd argument is not string type or const string type.", name);
+                time_zone_col = checkAndGetColumn<ColumnString>(time_zone_const_col->getDataColumn());
+                if (!time_zone_col)
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 2nd argument must be string type or const string type.", name);
+            }
```

Comment:
```suggestion
            const auto * time_zone_col = checkAndGetColumn<ColumnString>(arg2.column.get());
            const auto * time_zone_col_const = checkAndGetColumnConstData<ColumnString>(arg2.column.get());
            if (!time_zone_col && !time_zone_col_const)
                throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Illegal column {} of second argument of function {}. Expected String or Const(String)", arg2.column->getName(), name);
```
Note that the size of `time_zone_col_const` will be 1, so we should process it separately. Current implementation doesn't work correctly when first column is not const and the second is:
```
avogar-dev :) select toUTCTimestamp(materialize(toDateTime('2020-01-01')), 'Asia/Shanghai') from numbers(10)

0 rows in set. Elapsed: 0.001 sec.

Received exception:
Code: 1001. DB::Exception: std::bad_alloc: std::bad_alloc. (STD_EXCEPTION)
```

And maybe we don't need this if we want to allow only const time zone, in this case just work use `checkAndGetColumnConstData`, extract single timezone from it and use it further